### PR TITLE
[#6][#7] As a user, I can sign-in using my email and password | Reorganize the code by bussiness domain

### PR DESCRIPTION
**Pull request by longnd**
#7 

## What happened

This PR re-organizes the account-related code by [business domain](https://nimblehq.co/compass/development/code-conventions/elixir-phoenix/#business-domain) to follow Nimble's convention.
 
## Insight

- Using Context to group the account-related code. 
- Move the User module under `/schemas`
- Also update Phoenix & the other dependencies to the latest versions.
- Split the Test workflow into 3 steps: linting, unit test & feature test
 
## Proof Of Work

Screen Shot 2021-09-07 at 07 11 04

### CODE REVIEW

File: test/support/conn_case.ex [hunk 1]
```diff
@@ -26,6 +26,7 @@ defmodule GscraperWeb.ConnCase do
       import Phoenix.ConnTest
       import GscraperWeb.ConnCase
       import Gscraper.Factory
+      import GscraperWeb.Gettext
```
Comment (rosle) [inline test/support/conn_case.ex:29]: Maybe

```suggestion
      import Gscraper.Factory
      import GscraperWeb.ConnCase
      import GscraperWeb.Gettext
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -25,21 +29,26 @@ jobs:
...
 
-      - name: Setup Elixir
-        uses: erlef/setup-beam@v1
+      - name: Set up Elixir
+        uses: erlef/setup-elixir@v1
         with:
           otp-version: ${{ env.OTP_VERSION }}
           elixir-version: ${{ env.ELIXIR_VERSION }}
 
+      - name: Setup Node
```
Comment (rosle) [inline .github/workflows/test.yml:47]: Maybe fixing small wording here to make it consistent - `Set up` and `Setup`

**Review by rosle — APPROVED** [review]
👍

**Review by hanam1ni — APPROVED** [review]

File: .github/workflows/test.yml (hunk 2 above)
Comment (longnd) [reply .github/workflows/test.yml]: Thanks for catching it. Fixed in 111d23f

File: test/support/conn_case.ex (hunk 1 above)
Comment (longnd) [reply test/support/conn_case.ex]: yes, it must be like that. Fixed in deb8e19

**Pull request merged**